src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -520,8 +531,8 @@ internal void StartOperationCommon(Socket socket, SocketAsyncOperation operation
                 _context = null;
             }
 
-            // Capture execution context if none already.
-            if (_context == null)
+            // Capture execution context if necessary.
```

Comment:
Given that  Capture is cheap, and we're not even going to do it in the Task case, I think we can simplify the code above too: remove _completedChanged and just always store the socket.

Reply:
But if you want to punt for this PR, that's fine too...

Reply:
That's reasonable.  I'll change it separately.